### PR TITLE
feat(platform,core): custom title and subtitle template for dynamic page

### LINKED ISSUE #9929: [Sourcing][High][Jun] Need some control over subtitle in dynamic page
#### Is this a bug, enhancement, or feature request?

Enhancement

#### Briefly describe your proposal.

We want a control to render custom template in subtitle (shown below in the screenshot) of fundamental core dynamic page.

Screenshot 2023-05-26 at 4 24 23 PM

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)

@fundamental-ngx/core 0.41.0-rc.3
@fundamental-ngx/i18n 0.41.0-rc.3
@fundamental-ngx/platform 0.41.0-rc.3
@angular/cli 15.2.8

### PR DESCRIPTION
**Pull request by fkolar**
## Related Issue(s)

closes #9929

## Description

Introducing custom _subtitle template_ that can be added to the `DynamicPage` header. this will allow application developer to customize and extends subtitle 

## Screenshots

Screenshot 2023-06-22 at 11 26 22

##### During Implementation

1. Visual Testing:

-   [x] visual misalignments/updates
-   [x] check Light/Dark/HCB/HCW themes
-   [x] RTL/LTR - proper rendering and labeling
-   [x] responsiveness(resize)
-   [x] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [x] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [x] different combinations of components - free style
-   [x] change the API values during testing

3. Documentation and Example validations

-   [x] missing API documentation or it is not understandable
-   [x] poor examples
-   [x] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [x] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [x] Run npm run build-pack-library and test in external application
-   [x] update `README.md`
-   [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

File: libs/core/src/lib/dynamic-page/dynamic-page-header/header/dynamic-page-header.component.spec.ts [hunk 1]
```diff
@@ -35,6 +35,38 @@ class TestComponent {
...
+                <fd-breadcrumb-item>
+                    <a fd-link [attr.href]="'#'">Shoes</a>
+                </fd-breadcrumb-item>
+            </fd-breadcrumb>
+
+            <fd-dynamic-page-global-actions></fd-dynamic-page-global-actions>
+            <fd-dynamic-page-title-content></fd-dynamic-page-title-content>
+        </fd-dynamic-page-header>
+
+        <ng-template #subtitleTemplate>
```
Comment (N1XUS) [inline libs/core/src/lib/dynamic-page/dynamic-page-header/header/dynamic-page-header.component.spec.ts:54]: I would introduce some sort of structural directive inside content projection of `fd-dynamic-page-header` so developers wouldn't need to pass a reference as an input property of the component.

Comment (g-cheishvili) [reply libs/core/src/lib/dynamic-page/dynamic-page-header/header/dynamic-page-header.component.spec.ts:54]: +1 for this @fkolar

Comment (fkolar) [reply libs/core/src/lib/dynamic-page/dynamic-page-header/header/dynamic-page-header.component.spec.ts:54]: @N1XUS, This was my earlier discussion.  what is the preference of passing custom template and the response was F-NGX uses @Input and that F-NGX is consistent in this => @mikerodonnell89 

If its directive/component/ng-template  - this was my point. 

My preference would be simple ng-template.

**Comment by fkolar** [comment]
Can you guys re-check? @N1XUS , @mikerodonnell89

**Review by g-cheishvili — APPROVED** [review]

**Pull request merged**